src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.tsx

```diff
@@ -0,0 +1,344 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { Grid, Box, MenuItem, Select, Typography, Theme, SelectChangeEvent } from '@mui/material';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { CustomAGGrid, useSnackMessage } from '@gridsuite/commons-ui';
+import { fetchAllCountries } from '../../../../../../services/study/network-map';
+import { IdentifiableAttributes, evaluateJsonFilter } from '../../../../../../services/study/filter';
+import { fetchVoltageLevelsMapInfos } from '../../../../../../services/study/network';
+import CheckboxAutocomplete from '../../../../../utils/checkbox-autocomplete';
+import { useLocalizedCountries } from '../../../../../utils/localized-countries-hook';
+import { buildExpertFilter, CURVE_EQUIPMENT_TYPES, NOMINAL_VOLTAGE_UNIT } from './curve-selector-utils';
+import { EQUIPMENT_TYPES, VoltageLevel } from 'components/utils/equipment-types';
+import { AppState } from 'redux/reducer';
+import { AgGridReact } from 'ag-grid-react';
+
+export interface GetSelectedEquipmentsHandle {
+    api: {
+        getSelectedEquipments: () => IdentifiableAttributes[];
+    };
+}
```

Comment:
this interface is similar to GetSelectedItemsHandler. maybe have only one with 'getSelectedVariables' from GetSelectedItemsHandler is optional

Reply:
GetSelectedItemsHandler is not really good with useImperativeHandler, another word, we expose a set of methods (or states) which allows another component to manipulate inside states of the component.

I have proposed EquipmentFilterApi which is more general..

https://github.com/gridsuite/gridstudy-app/pull/2418/files#diff-2aba2d84e81110e5b0193f5156dca9400b60d94cb698f4e4418df2fe74f2e4a0R23-R24